### PR TITLE
RCS1246 - update, added Last

### LINKED ISSUE #1426: Add performance refactoring for IList Last, First or ElementAt
**Product and Version Used**: 
Visual Studio Professional 2022, 17.8.3
.Net 6.0
Rosslynator extension, 4.11.0

Index access versus list methods context: https://rules.sonarsource.com/csharp/RSPEC-6608/

**Steps to Reproduce**:
1. Populate a list, `someList`.
2. Access the first element through `someList.First()`.
3. Access the third element through `someList.ElementAt(2)`.
4. Access the first element through `someList.Last()`.

**Actual Behavior**:
No suggestions.

**Expected Behavior**:
Suggestions:
```csharp
someList[0];
someList[2];
someList[^1];
```

### PR DESCRIPTION
**Pull request by jakubreznak**
fixes #1426

### CODE REVIEW

File: ChangeLog.md [hunk 1]
```diff
@@ -11,6 +11,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 - Fix analyzer [RCS1077](https://josefpihrt.github.io/docs/roslynator/analyzers/RCS1077) ([PR](https://github.com/dotnet/roslynator/pull/1428))
 
+### Changed
+
+- Update - add Last [RCS1246](https://josefpihrt.github.io/docs/roslynator/analyzers/RCS1246) ([PR](https://github.com/dotnet/roslynator/pull/1431))
```
Comment (josefpihrt) [inline ChangeLog.md:16]: Fix PR number

File: src/Analyzers/CSharp/Analysis/InvocationExpressionAnalyzer.cs [hunk 2]
```diff
@@ -192,6 +192,23 @@ private static void AnalyzeInvocationExpression(SyntaxNodeAnalysisContext contex
                                 break;
                             }
                         case "Last":
+                            {
+                                if (DiagnosticRules.OptimizeLinqMethodCall.IsEffective(context))
+                                {
+                                    if (!invocationInfo.Expression.IsKind(SyntaxKind.ElementAccessExpression, SyntaxKind.InvocationExpression)
+                                        && UseElementAccessAnalysis.IsFixableLast(invocationInfo, context.SemanticModel, context.CancellationToken))
```
Comment (josefpihrt) [inline src/Analyzers/CSharp/Analysis/InvocationExpressionAnalyzer.cs:199]: Method `IsFixableLast` should contain a check if current compilation supports C# 8.0 or higher

**Review by josefpihrt — CHANGES_REQUESTED** [review]

**Review by josefpihrt — APPROVED** [review]

**Comment by josefpihrt** [comment]
👍

**Comment by josefpihrt** [comment]
Thanks!

**Comment by KavenBreton-eaton** [comment]
This is not usable for .Net Framework projects. Was this intended?

**Comment by josefpihrt** [comment]
@KavenBreton-eaton Could you please report it as a bug?

**Pull request merged**